src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -675,6 +675,14 @@ public String resolveIndexRouting(@Nullable String routing, String aliasOrIndex)
 
             aliasesOrIndices = convertFromWildcards(aliasesOrIndices, indicesOptions);
         }
+
+        if (isAllIndices(aliasesOrIndices)) {
+            if (!indicesOptions.allowNoIndices()) {
+                throw new IndexMissingException(new Index("_all"));
+            }
+            else return Strings.EMPTY_ARRAY;
+        }
+
```

Comment:
When both indicesOptions.expandWildcardsOpen() and indicesOptions.expandWildcardsClosed() are both false for some reason, concrete indices falls through to here. If the 'aliasOrIndices' is an '_all' pattern we want to return here. Otherwise the existing implementation runs into a Null Pointer a few lines below

```
if (aliasesOrIndices.length == 1)
```
At least that's what happens in the randomized test for this method when I introduced this. It's an edge case and the nesting of if-statements here is not nice, but there are so many different cases depending on the indicesOptions settings.